### PR TITLE
WIP: ENG-1414 - protos: Pipeline notification refactor

### PR DESCRIPTION
**Pull request by blinktag**
* Deprecated `Pipeline.Notify` boolean
* Deprecated `Pipeline.XNotificationConfigs`
* Dperecated `external.AttachNotification()` and `external.DetachNotification()`
* Add `PipelineStepCondition. PipelineStepNotification` to hold notification ID and payload inclusion settings

### CODE REVIEW

**Comment by dselans** [comment]
Need to rebase this against latest upstream + regen - added all the on_true/false stuff.

**Comment by blinktag** [comment]
Merged current main and resolved conflicts

**Comment by blinktag** [comment]
Let's wait to merge this until condition/response changes are pushed in SDKs

**Comment by dselans** [comment]
OK, SDK, server, demo-client, protos are all updated. Feel free to rebase and merge this in.

**Comment by dselans** [comment]
@blinktag Dunno if you already did it or not but remember to regenerate protos.

File: libs/protos/protos/sp_external.proto [hunk 1]
```diff
@@ -59,10 +59,14 @@ service External {
   rpc GetNotification(GetNotificationRequest) returns (GetNotificationResponse);
 
   // Attach a notification config to a pipeline
-  rpc AttachNotification(AttachNotificationRequest) returns (protos.StandardResponse);
+  rpc AttachNotification(AttachNotificationRequest) returns (protos.StandardResponse) {
+    option deprecated = true;
```
Comment (dselans) [inline libs/protos/protos/sp_external.proto:63]: Is this for deprecating methods? If so - can you stick a comment in there with a date noting why it was deprecated?

File: libs/protos/protos/sp_pipeline.proto [hunk 2]
```diff
@@ -45,15 +45,51 @@ enum AbortCondition {
...
+  // every time and also because that config info is sensitive and is encrypted
+  repeated string notification_config_ids = 1;
+
+  PayloadType payload_type = 2;
+
+  // If type == paths, then we will look here for a list of json paths to include
+  // in the notification payload.
+  repeated string paths = 3;
+
+  // Notification configs for this pipeline. Only filled out in external API responses
```
Comment (dselans) [inline libs/protos/protos/sp_pipeline.proto:89]: Not following the intent of this - remove if it's not needed / keep if you're still thinking it over or we can discuss in next standup.

**Review by dselans — APPROVED** [review]
Looks good. Remember to regenerate (if you haven't already).

File: libs/protos/protos/sp_pipeline.proto (hunk 2 above)
Comment (blinktag) [reply libs/protos/protos/sp_pipeline.proto:92]: It's deprecated, it was previously used for including the actual notification configuration, such as slack config or email config, in the response for frontend

File: libs/protos/protos/sp_external.proto (hunk 1 above)
Comment (blinktag) [reply libs/protos/protos/sp_external.proto:59]: Done!

**Pull request merged**